data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -226,8 +226,10 @@ private CloseableIterable<T> applyPosDeletes(CloseableIterable<T> records) {
 
     // if there are fewer deletes than a reasonable number to keep in memory, use a set
     if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < setFilterThreshold) {
```

Comment:
What about the else case? In that case we will not cache.

Reply:
Yes, it won't cache because the deletes are streamed from files.

Reply:
Yes, Because it use sort merge to avoid loading too many delete rows in memory.

Reply:
But, They should't reopen file and read file from head, they should read file continue。I will do some change here